Code/GraphMol/Descriptors/EEM.cpp

```diff
@@ -0,0 +1,237 @@
+// in the publication they don't have access to "Aromatic type"
+double getAtomtype(const RDKit::ROMol mol, const RDKit::Atom *atom) {
+      double t=1.0;
+      double a;
+      RDKit::ROMol::ADJ_ITER nbrIdx, endNbrs;
+      boost::tie(nbrIdx, endNbrs) = mol.getAtomNeighbors(atom);
+      while (nbrIdx != endNbrs) {
+        const RDKit::Bond *bond = mol.getBondBetweenAtoms(*nbrIdx, atom->getIdx());
+        a = bond->getBondTypeAsDouble();
+        if (a == 1.5) {
+             t = 2.0;
+            }
+        if (a == 2.0) {
+             t = 2.0;
+            }
+        if (a == 3.0) {
+             t = 3.0;
+            }
+        ++nbrIdx;
+  }
+  return t;
+}
+
+double* getEEMMatrix(RDKit::ROMol mol, double * dist3D, int n, EEM_arrays EEMatoms) {
+  int sizeArray = (n+1) * (n+1);
+  double* EEM = new double[sizeArray](); // declaration to set all elements to zeros!
+  #define IDX(x, y) (x * (n+1) + y)
+    /* Fill the full n * n block */
+    for(long int i = 0; i < n; i++) {
+        double t = EEMatoms.EEMatomtype[i];
+        int idx = EEMatoms.Atomindex[i];
+        double v;
+        if (t== 1.0) {
+                v=  B1[idx];
+        }
+        if (t== 2.0) {
+                v = B2[idx];
+        }
+        if (t== 3.0) {
+                v = B3[idx];
+        }
+
+        EEM[IDX(i, i)] = v;
+        for(long int j = i + 1; j < n; j++) {
+                EEM[IDX(i, j)] = kappa / dist3D[i*n+j];
+                EEM[IDX(j, i)] = EEM[IDX(i, j)];
+            }
+        }
+    /* Fill last column & row */
+    for(long int i = 0; i < n; i++) {
+        EEM[IDX(n, i)] = 1.0; // column
+        EEM[IDX(i, n)] = -1.0; // row
+
+    }
+    #undef IDX
+    return EEM;
+}
+
+double* getBVector(RDKit::ROMol mol, int n, EEM_arrays EEMatoms) {
+       /* Fill vector b i.e. -A */
+        double * b = new double[n+1];
+        for(int j = 0; j < n; j++) {
+          double t = EEMatoms.EEMatomtype[j];
+          int idx = EEMatoms.Atomindex[j];
+            if ( t== 1.0) {
+                b[j] = -A1[idx];
+            }
+            if (t== 2.0) {
+                b[j] = -A2[idx];
+            }
+            if (t== 3.0) {
+                b[j] = -A3[idx];
+            }
+        }
+
+        b[n] = RDKit::MolOps::getFormalCharge(mol); // sum of charges
+
+    return b;
+}
+
+
+EEM_arrays calculate_EEM_parameters(RDKit::ROMol mol, int n) {
+       /* Fill vector b i.e. -A */
+        int * a = new int[n];
+        double * b = new double[n];
+
+        for(int j = 0; j < n; j++) {
+            b[j]=getAtomtype(mol,mol.getAtomWithIdx(j));
+            a[j]=mol.getAtomWithIdx(j)->getAtomicNum();
+        }
+
+    return EEM_arrays {a,b};
+}
+
+
+/* Calculate charges for a particular kappa_data structure */
+std::vector<double> calculate_charges(RDKit::ROMol mol, double * dist3D, int numAtoms, EEM_arrays EEMatoms) {
+        double * A = new double[(numAtoms+1)*(numAtoms+1)];
```

Comment:
you allocate the memory here and then immediately throw it away in the next line. 
You don't need the `new` here.